test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,395 @@
+    private volatile static boolean componentMoved = false;
+    private volatile static boolean componentResized = false;
+    private volatile static Point compAt;
+    private volatile static Dimension compSize;
+    private volatile static ArrayList<ComponentEvent> events =
+        new ArrayList<>();
+
+    private static final ComponentListener componentListener =
+        new ComponentListener() {
+
+            @Override
+            public void componentShown(ComponentEvent e) {
+                System.out.println("ComponentShown: " + e.getSource());
+                componentShown = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentResized(ComponentEvent e) {
+                System.out.println("ComponentResized: " + e.getSource());
+                componentResized = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentMoved(ComponentEvent e) {
+                System.out.println("ComponentMoved: " + e.getSource());
+                componentMoved = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+                System.out.println("ComponentHidden: " + e.getSource());
+                componentHidden = true;
+                events.add(e);
+            }
+        };
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.delay(DELAY);
+            robot.waitForIdle();
```

Comment:
```suggestion
            robot.waitForIdle();
            robot.delay(DELAY);
```
Delaying before `waitForIdle` doesn't make much sense… The tests use `waitForIdle` followed by `delay` to ensure all the events are handled and then to add additional time… just in case.

If you call `waitForIdle` after `delay`, the chances are high that `waitForIdle` returns immediately because there are no pending events already (they're handled when the thread has been sleeping).